File: spring-mvc/src/main/java/software/darkmatter/school/blog/domain/post/business/PostService.java
package software.darkmatter.school.blog.domain.post.business;

import org.springframework.data.domain.Pageable;
import software.darkmatter.school.blog.api.dto.PostCreateDto;
import software.darkmatter.school.blog.domain.post.data.Post;

import java.util.List;

public interface PostService {

    List<Post> getList(Pageable pageable);

    Post getById(Long id);

    Post create(PostCreateDto postCreateDto);

    Post update(Long id, PostCreateDto postCreateDto);

    void delete(Long id);

    void publish(Long id);
}

File: kotlin-spring-webflux/src/main/kotlin/software/darkmatter/school/blog/domain/post/business/PostService.kt
package software.darkmatter.school.blog.domain.post.business

import org.springframework.data.domain.Pageable
import software.darkmatter.school.blog.api.dto.PostCreateDto
import software.darkmatter.school.blog.api.dto.PostPublishDto
import software.darkmatter.school.blog.domain.post.data.Post

interface PostService {
    suspend fun getList(pageable: Pageable): List<Post>
    suspend fun getById(id: Long): Post
    suspend fun create(postCreateDto: PostCreateDto): Post
    suspend fun update(id: Long, postCreateDto: PostCreateDto): Post
    suspend fun delete(id: Long)
    suspend fun publish(id: Long, body: PostPublishDto)
}
